### PR TITLE
Workaround for analyzerconfig + sg crash:

### PR DESCRIPTION
**Pull request by chsienki**
- Workaround analyzer config options not being available for generated trees by getting new options for all trees after generation
- Add test to cover crash + fix

This is a band-aid fix for #44087 

Eventually we'll want to actually parse the generated trees with the `AnalyzerConfigSet` and when doing that we can return the options at the same time, meaning we won't do any duplicate work. That requires threading a whole bunch of stuff through the generator though, and I wanted to get a fix in first that fixed the compiler crash.

### CODE REVIEW

**Comment by chsienki** [comment]
I've opened https://github.com/dotnet/roslyn/issues/44223 to track doing the real work

**Comment by chsienki** [comment]
Ah, this breaks something else :( . Investigating.

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review. 

/cc @jaredpar @agocke

**Comment by chsienki** [comment]
Close / reopen to kick CI

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please. I'd like to get this in for 16.7p2

File: src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs [hunk 1]
```diff
@@ -788,6 +788,16 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
             // We pass it to the generators, which will realize any symbols they require. 
             compilation = RunGenerators(compilation, Arguments.ParseOptions, generators, additionalTexts, diagnostics);
 
+            // https://github.com/dotnet/roslyn/issues/44087 
+            // Workaround by re-getting options for *all* the syntax trees including generated ones if there were
+            // any generators to run. We'll actually want to parse the generated files according to the rules of the config set
+            // and add them in here.
+            if (Arguments.AnalyzerConfigPaths.Length > 0 && generators.Length > 0)
```
Comment (cston) [inline src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:795]: >Arguments.AnalyzerConfigPaths.Length [](start = 16, length = 36)

Should this be `!sourceFileAnalyzerConfigOptions.IsDefault`?

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 2]
```diff
@@ -12352,4 +12372,32 @@ public override void CreateAnalyzerWithinCompilation(CompilationStartAnalysisCon
...
+
+    [Generator]
+    internal class SimpleGenerator : ISourceGenerator
+    {
+        private readonly string _sourceToAdd;
+
+        public SimpleGenerator()
+        {
+            _sourceToAdd = string.Empty;
+        }
```
Comment (cston) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:12384]: Minor: Consider removing if not used.

Comment (chsienki) [reply src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:12384]: It's used implicitly by other tests. There are some that create anything they can find via reflection, so they need to have a parameter less ctor. I'll add a comment though to explain it.

File: src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs (hunk 1 above)
Comment (chsienki) [reply src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:795]: Either works, but that's probably better.

File: src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs [hunk 3]
```diff
@@ -788,6 +788,16 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
             // We pass it to the generators, which will realize any symbols they require. 
             compilation = RunGenerators(compilation, Arguments.ParseOptions, generators, additionalTexts, diagnostics);
 
+            // https://github.com/dotnet/roslyn/issues/44087 
+            // Workaround by re-getting options for *all* the syntax trees including generated ones if there were
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:792]: Is the `all` portion of this statement still true?

Comment (chsienki) [reply src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs:792]: Derp. Fixed.

**Pull request merged**